op-batcher/batcher/espresso.go

```diff
@@ -21,43 +22,529 @@ import (
+// for transaction submissions.
+type espressoTransactionJobAttempt struct {
+	job  espressoSubmitTransactionJob
+	resp chan espressoSubmitTransactionJobResponse
+}
+
+// espressoVerifyReceiptJob is a struct that holds the state required to
+// verify a receipt for a transaction that was submitted to Espresso.
+// It contains the transaction that was submitted, the hash of the
+// transaction, and the number of attempts to verify the receipt.
+type espressoVerifyReceiptJob struct {
+	attempts    int
+	start       time.Time
+	transaction espressoSubmitTransactionJob
+	hash        *espressoCommon.TaggedBase64
+}
+
+// espressoVerifyReceiptJobResponse is a struct that holds the
+// response from the Espresso client after verifying a receipt.
+// It contains the job that was submitted, and any error that occurred
+// during the verification (if unsuccessful).
+type espressoVerifyReceiptJobResponse struct {
+	job espressoVerifyReceiptJob
+	err error
+}
+
+// espressoVerifyReceiptJobAttempt is a struct that holds the job and
+// response channel for a receipt verification job.
+//
+// This is the unit of work that is submitted to the worker to process
+// for receipt verifications.
+type espressoVerifyReceiptJobAttempt struct {
+	job  espressoVerifyReceiptJob
+	resp chan espressoVerifyReceiptJobResponse
+}
+
+// espressoTransactionSubmitter is a struct that holds the state that governs
+// the worker queue processing details for submitting transactions to Espresso
+// without spawning arbitrarily many goroutines.
+type espressoTransactionSubmitter struct {
+	ctx                      context.Context
+	wg                       *sync.WaitGroup
+	submitJobQueue           chan espressoSubmitTransactionJob
+	submitRespQueue          chan espressoSubmitTransactionJobResponse
+	submitWorkerQueue        chan chan espressoTransactionJobAttempt
+	verifyReceiptJobQueue    chan espressoVerifyReceiptJob
+	verifyReceiptRespQueue   chan espressoVerifyReceiptJobResponse
+	verifyReceiptWorkerQueue chan chan espressoVerifyReceiptJobAttempt
+	espresso                 espressoClient.EspressoClient
+}
+
+// EspressoTransactionSubmitterConfig is a configuration struct for the
+// EspressoTransactionSubmitter. It contains the configurable details for
+// creating the EspressoTransactionSubmitter.
+type EspressoTransactionSubmitterConfig struct {
+	Ctx                                context.Context
+	EspressoClient                     espressoClient.EspressoClient
+	Wg                                 *sync.WaitGroup
+	SubmitJobQueueCapacity             int
+	SubmitResponseQueueCapacity        int
+	VerifyReceiptJobQueueCapacity      int
+	VerifyReceiptResponseQueueCapacity int
+}
+
+// EspressoTransactionSubmitterOption is a function that can be used to
+// configure the EspressoTransactionSubmitterConfig.
+type EspressoTransactionSubmitterOption func(*EspressoTransactionSubmitterConfig)
+
+// WithEspressoClient is an option that can be used to set the Espresso client
```

Comment:
```suggestion
// WithContext is an option that can be used to set the Espresso client
```

Reply:
Addressed in db23892adc18ad7fe5ff7c51cfffb1203b8cb3ac.